src/main/java/com/xceptance/xlt/report/providers/ErrorsReportProvider.java

```diff
@@ -465,42 +465,35 @@ public void processDataRecord(final Data stat)
                     final String directoryHint = txnStats.getDumpDirectoryPath();
                     if (directoryHint != null)
                     {
+                        // check if we should add/replace a result browser directory hint
                         final int size = errorReport.directoryHints.size();
-                        if (size < directoryLimitPerError)
+                        if (size < directoryLimitPerError || random.nextDoubleFast() <= directoryReplacementChance)
```

Comment:
Small optimization: extract the expression `size < directoryLimitPerError` into a local variable, e.g. `safeToAdd`, and re-use it here and at line 481.

Reply:
Done.